pubsublite-kafka-shim/src/main/java/com/google/cloud/pubsublite/kafka/KafkaExceptionUtils.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.kafka;
+
+import com.google.cloud.pubsublite.internal.ExtractStatus;
+import io.grpc.StatusException;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.AuthenticationException;
+import org.apache.kafka.common.errors.AuthorizationException;
+import org.apache.kafka.common.errors.BrokerNotAvailableException;
+import org.apache.kafka.common.errors.InterruptException;
+import org.apache.kafka.common.errors.InvalidRequestException;
+import org.apache.kafka.common.errors.TopicExistsException;
+
+class KafkaExceptionUtils {
+  private KafkaExceptionUtils() {}
+
+  static KafkaException toKafkaException(StatusException source) {
+    switch (source.getStatus().getCode()) {
+      case OK:
+        throw source.getStatus().asRuntimeException();
+      case ABORTED:
+        return new BrokerNotAvailableException("Aborted.", source);
+      case ALREADY_EXISTS:
+        return new TopicExistsException("", source);
```

Comment:
I'd be inclined to just make this a raw KafkaException, since there is no guarantee that the NOT_FOUND error is for a topic (and not a subscription).

Reply:
Done.